CMakeLists.txt

```diff
@@ -44,23 +45,47 @@ target_compile_definitions(GSL INTERFACE
 
 # add include folders to the library and targets that consume it
 target_include_directories(GSL INTERFACE
-    $<BUILD_INTERFACE:
-        ${CMAKE_CURRENT_SOURCE_DIR}/include
-    >
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
 )
 
 # add natvis file to the library so it will automatically be loaded into Visual Studio
 target_sources(GSL INTERFACE
-    ${CMAKE_CURRENT_SOURCE_DIR}/GSL.natvis
-)
-
-install(
-    DIRECTORY include/gsl
-    DESTINATION include
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/GSL.natvis>
 )
 
 option(GSL_TEST "Generate tests." ${GSL_STANDALONE_PROJECT})
 if (GSL_TEST)
 	enable_testing()
 	add_subdirectory(tests)
 endif ()
+
+set(package_name "MicrosoftGSL")
+set(targets_export_name "${package_name}Targets")
+set(project_config "${package_name}Config.cmake")
+set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${package_name}")
+set(project_namespace "${package_name}::")
+install(
```

Comment:
What would you think about guarding this installs behind a variable?
AFAIR not guarding turns into a problem when people consume the library as a git submodule and their own product has install directives. So if you run install in the parent repo gsl also get installed which might not be wanted.

If you think this is out of scope of this change I can open an issue

Reply:
If the parent doesn't want to install GSL, they should add [`EXCLUDE_FROM_ALL`](https://cmake.org/cmake/help/latest/command/add_subdirectory.html) to the `add_subdirectory` call.

Reply:
That would work, but I am not sure it is the best way. Here are some drawbacks:
https://stackoverflow.com/questions/16486785/cmake-exclude-custom-install-targets-from-make-install-all
Anyway not a big issue

Reply:
thanks for comments. In current CMake solution `install` isn't guarded with variable. If it's needed I can rise seperate PR.

Reply:
Another option would be to specify `EXCLUDE_FROM_ALL` and a `COMPONENT` for `install` if this isn't the top-level CMakeLists.txt